kani-compiler/src/codegen_cprover_gotoc/codegen/statement.rs

```diff
@@ -390,6 +390,11 @@ impl<'tcx> GotocCtx<'tcx> {
                 }
             })
             .collect();
+
+        if self.is_intrinsic(func) {
```

Comment:
is `is_instrinsic` just duplicating a bunch of code that's already right below these lines?

Couldn't we just inline the `if matches!...` part of the above function below after the `let instance`?